### PR TITLE
Allow Broker apps to acquire token through ad-accounts

### PR DESCRIPTION
**Pull request by rapong**
Remove the check which prevents Broker app to acquire token through broker.

This is required for the AADNGC (Work account password less sign-in) scenarios.

### CODE REVIEW

File: adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java [hunk 1]
```diff
@@ -634,6 +634,11 @@ private void verifyBrokerRedirectUri(final AuthenticationRequest request) throws
             throw new UsageAuthenticationException(ADALError.DEVELOPER_REDIRECTURI_INVALID, "The redirectUri is null or blank.");
         }
 
+        if (inputUri.equalsIgnoreCase(AuthenticationConstants.Broker.BROKER_REDIRECT_URI))
+        {
```
Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java]: Nit: our bracket-style is Egyptian

Can we add a log statement here to capture the early-return?

File: adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java [hunk 2]
```diff
@@ -124,8 +124,6 @@ public SwitchToBroker canSwitchToBroker(final String authorityUrlStr) {
         }
         final String packageName = mContext.getPackageName();
         boolean canSwitchToBroker = AuthenticationSettings.INSTANCE.getUseBroker()
-                && !packageName.equalsIgnoreCase(AuthenticationSettings.INSTANCE.getBrokerPackageName())
```
Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java]: We'll need to get Company Portal to test out these changes too, but can merge first and be reintroduced/modified later as needed

Comment (rapong) [reply adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java]: Added a comment on top of the function as a reminder.

File: adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java [hunk 3]
```diff
@@ -110,9 +110,12 @@
      * Verifies the broker related app and AD-Authenticator in Account Manager
      * ADAL directs call to AccountManager if component is valid and present. It
      * does not direct call if the caller is from Authenticator itself.
```
Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java]: This documentation is now incorrect, right? Can you now clarify the expected functionality here?

**Comment by rapong** [comment]
@iambmelt needs a sign-off here as well. Thanks!

**Comment by iambmelt** [comment]
@rapong Reviewing now, cheers 👍

**Pull request merged**